pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -812,3 +908,180 @@ func loadBalancerSourceRangesMatch(ic *operatorv1.IngressController, current *co
+		return false
+	}
+
+	// Create maps to track the IDs from each subnet object for comparison.
+	subnetIDMap1 := make(map[operatorv1.AWSSubnetID]bool)
+	subnetIDMap2 := make(map[operatorv1.AWSSubnetID]bool)
+	for _, id := range subnets1.IDs {
+		subnetIDMap1[id] = true
+	}
+	for _, id := range subnets2.IDs {
+		subnetIDMap2[id] = true
+	}
+	// Check if maps contain the same IDs.
+	for id := range subnetIDMap1 {
+		if !subnetIDMap2[id] {
+			return false
+		}
+	}
+	// Create maps to track the names from each subnet object for comparison.
+	subnetNameMap1 := make(map[operatorv1.AWSSubnetName]bool)
+	subnetNameMap2 := make(map[operatorv1.AWSSubnetName]bool)
+	for _, name := range subnets1.Names {
+		subnetNameMap1[name] = true
+	}
+	for _, name := range subnets2.Names {
+		subnetNameMap2[name] = true
+	}
+	// Check if maps contain the same names.
+	for name := range subnetNameMap1 {
+		if !subnetNameMap2[name] {
+			return false
+		}
+	}
+
+	return true
+}
+
+// awsSubnetsExist checks if any subnets exist in an AWSSubnets structure.
+func awsSubnetsExist(subnets *operatorv1.AWSSubnets) bool {
+	return subnets != nil && (len(subnets.Names) > 0 || len(subnets.IDs) > 0)
+}
+
+// JoinAWSSubnets joins an AWS Subnets object into a string seperated by sep.
+func JoinAWSSubnets(subnets *operatorv1.AWSSubnets, sep string) string {
+	if subnets == nil {
+		return ""
+	}
+	joinedSubnets := ""
+	subnetCount := 0
+	for _, subnet := range subnets.IDs {
+		if subnetCount > 0 {
+			joinedSubnets += sep
+		}
+		joinedSubnets += string(subnet)
+		subnetCount++
+	}
+	for _, subnet := range subnets.Names {
+		if subnetCount > 0 {
+			joinedSubnets += sep
+		}
+		joinedSubnets += string(subnet)
+		subnetCount++
+	}
+	return joinedSubnets
+}
+
+// getAWSLoadBalancerTypeInStatus gets the AWS Load Balancer Type reported in the status.
+func getAWSLoadBalancerTypeInStatus(ic *operatorv1.IngressController) operatorv1.AWSLoadBalancerType {
+	if ic.Status.EndpointPublishingStrategy != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil {
+		return ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type
+	}
+	return ""
+}
+
+// getAWSClassicLoadBalancerParametersInSpec gets the ClassicLoadBalancerParameter struct
+// defined in the IngressController spec.
+func getAWSClassicLoadBalancerParametersInSpec(ic *operatorv1.IngressController) *operatorv1.AWSClassicLoadBalancerParameters {
+	if ic.Spec.EndpointPublishingStrategy != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil {
+		return ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.ClassicLoadBalancerParameters
```

Comment:
I think you need to check nil for ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil this field is [optional](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L542) and is a pointer .